disnake/role.py

```diff
@@ -536,14 +602,39 @@ async def edit(
             await self._move(position, reason=reason)
 
         payload: Dict[str, Any] = {}
-        if color is not MISSING:
-            colour = color
 
-        if colour is not MISSING:
-            if isinstance(colour, int):
-                payload["color"] = colour
+        colors: Dict[str, Any] = {}
+        if color is not MISSING or colour is not MISSING:
+            primary_colour = color or colour
+
+        if primary_color is not MISSING:
+            primary_colour = color
+
+        if primary_colour is not MISSING:
+            if isinstance(primary_colour, int):
+                colors["primary_color"] = primary_colour
             else:
-                payload["color"] = colour.value
+                colors["primary_color"] = primary_colour.value
+
+        if secondary_color is not MISSING:
+            secondary_colour = secondary_color
+
+        if secondary_colour is not MISSING:
+            if isinstance(secondary_colour, Colour):
+                colors["secondary_color"] = secondary_colour.value
+            else:
+                colors["secondary_color"] = secondary_colour
+
+        if tertiary_color is not MISSING:
+            tertiary_colour = tertiary_color
+
+        if tertiary_colour is not MISSING:
+            if isinstance(tertiary_colour, Colour):
+                colors["tertiary_color"] = tertiary_colour.value
+            else:
+                colors["tertiary_color"] = tertiary_colour
+
+        payload["colors"] = colors
```

Comment:
On the topic of editing individual color fields: when calling `Role.edit(secondary_color=)`, one *might* expect the primary and tertiary colors to be kept as-is? Not sure if that's a reasonable assumption, though.

Assuming we do actually implement it that way, I'd imagine something like this could work:
```py
if primary_color is not MISSING or secondary_color is not MISSING or tertiary_color is not MISSING:
    colors: RoleColorsPayload = {"primary_color": self._primary_color, ...}
    if primary_color is not MISSING:
        colors["primary_color"] = ...
    # etc.
    payload["colors"] = colors
```

Reply:
Yeah, it's somewhat unnecessary, though depending on how Discord handles these things (which has been inconsistent), it might result in an audit log entry despite colors not actually being changed.

Reply:
https://github.com/DisnakeDev/disnake/pull/1312/commits/8db0056691912da83b96ca5504e1b9d79501aecf